### PR TITLE
Add external grid and clock setters.

### PR DESCRIPTION
**Pull request by WilliamJamieson**
## Description

Provides setting API for external grids and clocks as suggested by @rmontouro for the UFS GOCART component

## Motivation and Context

It allows a NUOPC cap to set the grid and clock instead of relying MAPL to create a grid and clock.

## How Has This Been Tested?

This works for the UFS NUOPC component
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Review by tclune — COMMENTED** [review]
I want to hold off a day on this.   @atrayano just committed a bug fix that we want to role out as a new release.  Unfortunately this was done on develop.   I don't want to keep rolling other changes into the release.   So we wait.  (notifying @mathomp4 too)

**Review by tclune — APPROVED** [review]

**Comment by mathomp4** [comment]
I'd like to wait until the `main` -> `develop` PR is done before we merge this.

File: CHANGELOG.md
Comment (mathomp4) [inline CHANGELOG.md:20]: Whoops. Looks like Git decided to add this to 2.3.5 in a merge maybe. @WilliamJamieson Can you move this up to the Unreleased section.

Comment (mathomp4) [reply CHANGELOG.md]: Okay I fixed this

**Review by tclune — APPROVED** [review]

**Pull request merged**
